### PR TITLE
Fix and improvements to aws creds

### PR DESCRIPTION
**Pull request by renkelvin**
@markdroth Please take a look and let me know if you have any questions. Thanks!

### CODE REVIEW

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc [hunk 1]
```diff
@@ -383,9 +384,12 @@ void AwsExternalAccountCredentials::BuildSubjectToken() {
   headers.push_back(Json({{"key", "host"}, {"value", signed_headers["host"]}}));
   headers.push_back(
       Json({{"key", "x-amz-date"}, {"value", signed_headers["x-amz-date"]}}));
+  headers.push_back(Json({{"key", "x-amz-security-token"},
+                          {"value", signed_headers["x-amz-security-token"]}}));
+  headers.push_back(
+      Json({{"key", "x-goog-cloud-target-resource"}, {"value", audience_}}));
```
Comment (renkelvin) [inline src/core/lib/security/credentials/external/aws_external_account_credentials.cc:390]: 2 more header expected by our backend.

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc [hunk 2]
```diff
@@ -383,9 +384,12 @@ void AwsExternalAccountCredentials::BuildSubjectToken() {
...
   Json::Object object{{"url", Json(cred_verification_url_)},
                       {"method", Json("POST")},
-                      {"body", Json("")},
```
Comment (renkelvin) [inline src/core/lib/security/credentials/external/aws_external_account_credentials.cc:388]: `body` will always be empty and won't be checked by our backend. Removed.

File: src/core/lib/security/credentials/external/aws_external_account_credentials.cc [hunk 3]
```diff
@@ -310,39 +311,39 @@ void AwsExternalAccountCredentials::OnRetrieveSigningKeysInternal(
     GRPC_ERROR_UNREF(error);
     return;
   }
-  auto it = json.object_value().find("access_key_id");
+  auto it = json.object_value().find("AccessKeyId");
```
Comment (renkelvin) [inline src/core/lib/security/credentials/external/aws_external_account_credentials.cc:314]: AWS production environment expects the key to be camel cased.

**Review by markdroth — APPROVED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/grpc/grpc/24990)* status: all files reviewed, 2 unresolved discussions (waiting on @renkelvin)

**Comment by renkelvin** [comment]
I think the CI failure should be a tooling flake. We should be safe to merge this.

**Comment by markdroth** [comment]
Known issue: #24596

**Pull request merged**